src/TimedSet.ts

```diff
@@ -18,198 +20,185 @@ export interface ITimedSet<T> extends Set<T> {
      * @param key - Key
      */
     refresh(key: T): boolean;
-}
 
+    /**
+     * Get the underlying set
+     */
+    uniqueSet(): Set<T>;
+}
 
 /**
  * This set will evict items from the array after the set timeout.
  * This set can only contain unique items, items are unique when === is true
  */
-export class TimedSet<T> implements ITimedSet<T> {
-    private _map: Map<T, Timer>;
+export class TimedSet<T> implements ITimedSet<T>, SetLike<T>, SetUtils<T> {
```

Comment:
why are you implementing ` SetLike<T>, SetUtils<T>`? Set in the ESNext spec is already defined to have these methods. plus, this PR removes `ITimedSet`.

Reply:
bc this doesn't extend Set, it just implements it. that said, we can just check if the current env supports the new Set operations and run it on the prototype directly